### PR TITLE
Sip2 81

### PR DESCRIPTION
**Pull request by Arghya22**
SIP2-81 ( https://issues.folio.org/browse/SIP2-81 ) 

#Purpose : 

- Charged items count (that is, total count of items currently loaned to the user)
- Fine items count (that is, total count of items with fines for this user) 

#Approach : 

- Corresponding API is integrated for fetching account & loan related data 
- Using above charged & fine item count are calculated for a specific user 
- Corresponding test cases added 

image

### CODE REVIEW

**Review by nirmalr-shah — APPROVED** [review]

**Review by arin-epm — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java [hunk 1]
```diff
@@ -159,6 +159,16 @@ private Future<PatronInformationResponse> validPatron(String userId, Personal pe
     final Future<PatronInformationResponseBuilder> manualBlocksFuture = feeFinesRepository
         .getManualBlocksByUserId(userId, sessionData)
         .map(blocks -> buildPatronStatus(blocks, builder));
+    // add fine count
+    final Future<PatronInformationResponseBuilder> accountFuture = feeFinesRepository
+        .getAccountDataByUserId(userId, sessionData)
+        .map(accounts -> populateFinesCount(accounts,builder));
+
+    //add charged count
```
Comment (steveellis) [inline src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:167]: Would fix this comment up a bit. Add space between "//: and "a". Capitalize "a".

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java [hunk 2]
```diff
@@ -159,6 +159,16 @@ private Future<PatronInformationResponse> validPatron(String userId, Personal pe
     final Future<PatronInformationResponseBuilder> manualBlocksFuture = feeFinesRepository
         .getManualBlocksByUserId(userId, sessionData)
         .map(blocks -> buildPatronStatus(blocks, builder));
+    // add fine count
```
Comment (steveellis) [inline src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:162]: Maybe capitalize "a" in the comment to make it conform to other comments in the code. It's super clean code! Might as well make the comments clean too.

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java [hunk 3]
```diff
@@ -159,6 +159,16 @@ private Future<PatronInformationResponse> validPatron(String userId, Personal pe
...
+    final Future<PatronInformationResponseBuilder> accountFuture = feeFinesRepository
+        .getAccountDataByUserId(userId, sessionData)
+        .map(accounts -> populateFinesCount(accounts,builder));
```
Comment (steveellis) [inline src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:165]: Would add a space between the params: `accounts,builder` should be `accounts, builder`.

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java [hunk 4]
```diff
@@ -159,6 +159,16 @@ private Future<PatronInformationResponse> validPatron(String userId, Personal pe
...
+
+    //add charged count
+    final Future<PatronInformationResponseBuilder> loansFuture = circulationRepository
+        .getLoansByUserId(userId, null, null, sessionData)
+        .map(loans -> populateChargedCount(loans,builder));
```
Comment (steveellis) [inline src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:170]: Same. Need space between params.

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java [hunk 5]
```diff
@@ -191,6 +200,28 @@ private Future<PatronInformationResponse> validPatron(String userId, Personal pe
         );
   }
 
+  private PatronInformationResponseBuilder populateChargedCount(JsonObject loans,
+                                      PatronInformationResponseBuilder builder) {
+    final int chargedItemsCount;
+    if (loans != null) {
+      chargedItemsCount = Math.min(getTotalRecords(loans), 9999);
```
Comment (steveellis) [inline src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:207]: This is interesting. What is `Math.min` is doing for us here? Does it need a comment perhaps?

**Review by steveellis — APPROVED** [review]
Forgive me if you mentioned it in the standup but assuming you've also tested against snapshot.

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java (hunk 1 above)
Comment (Arghya22) [reply src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java]: Added

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java (hunk 2 above)
Comment (Arghya22) [reply src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java]: Added

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java (hunk 3 above)
Comment (Arghya22) [reply src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java]: Added

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java (hunk 4 above)
Comment (Arghya22) [reply src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java]: Added

File: src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java (hunk 5 above)
Comment (Arghya22) [reply src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java:208]: Get minimum of total fine count & 9999 OR 
 Get minimum of total loan count & 9999

**Pull request merged**